src/include/daos_pool.h

```diff
@@ -288,7 +291,7 @@ daos_pool_global2local(d_iov_t glob, daos_handle_t *poh);
  *			-DER_NO_HDL	Invalid pool handle
  */
 int
-daos_pool_query(daos_handle_t poh, d_rank_list_t *tgts, daos_pool_info_t *info,
+daos_pool_query(daos_handle_t poh, d_rank_list_t **ranks, daos_pool_info_t *info,
```

Comment:
that's a major version API change? so probably has  to be done in a bwd compatible way.
that being said though, from how i used it, tgts was always set to NULL

Reply:
Yes I think an API version update is needed. I realized this after the push. Would it be a minor API version bump if the ABI is (roughly) the same (both pointer type arguments)? So changing from 2.1.0 to 2.2.0 API version in SConstruct (as opposed to 2.1.0 to 3.1.0)?

Not sure about how backward compatibility would be implemented in this type of API change scenario. But another consideration (suggesting I would guess not much if any back compatibility is needed in this instance) is that the existing implementation dc_pool_query() has D_ASSERT(args->tgts == NULL); /* TODO */.

Reply:
the thing is anyone who is using this API without a NULL argument would have to change his code, and recompile, no? that would be a major API change. but again as far as im aware, i have not used this API with the tgts not being NULL. so maybe this would be ok.

Reply:
@mchaarawi and I discussed and all known API users pass NULL as the tgts parameter so (based on what we see now) it doesn't seem like a back-compatibility mechanism is needed (i.e., a minor API bump with deprecation of daos_pool_query and addition of say a daos_pool_query2 function). And also does not seem that a major API bump is needed, that would otherwise be done with the current form of the PR.

There will be follow-on patches to come soon to fill out the usage of the new pool query functionality. If we discover something not considered so far, we can go down the back-compatibility + minor API bump track while we make those changes.

I will back out the minor API version bump in this PR. While I'm at it I will merge latest master and push once more to CI since some test issues have been resolved (a chance for a clean CI run perhaps).